src/frontend/utils/updateLiveParticipants.ts

```diff
@@ -0,0 +1,76 @@
+import { updateLiveParticipants } from 'data/sideEffects/updateLiveParticipants';
+import { Participant } from 'types/Participant';
+import { Video } from 'types/tracks';
+
+export const addParticipantAskingToJoin = async (
+  video: Video,
+  participant: Participant,
+) => {
+  const participants = {
+    participantsAskingToJoin: video.participants_asking_to_join.concat([
+      participant,
+    ]),
+    participantsInDiscussion: video.participants_in_discussion,
+  };
+  return await updateLiveParticipants(video, participants);
+};
+
+export const addParticipantToDiscussion = async (
+  video: Video,
+  participant: Participant,
+) => {
+  const participants = {
+    participantsAskingToJoin: video.participants_asking_to_join,
+    participantsInDiscussion: video.participants_in_discussion.concat([
+      participant,
+    ]),
+  };
+  return await updateLiveParticipants(video, participants);
+};
+
+export const removeParticipantAskingToJoin = async (
+  video: Video,
+  participant: Participant,
+) => {
+  if (!video.participants_asking_to_join.includes(participant)) return;
+
+  const participants = {
+    participantsAskingToJoin: video.participants_asking_to_join.filter(
+      (p) => p.id !== participant.id,
+    ),
+    participantsInDiscussion: video.participants_in_discussion,
+  };
+  return await updateLiveParticipants(video, participants);
+};
+
+export const removeParticipantFromDiscussion = async (
+  video: Video,
+  participant: Participant,
+) => {
+  if (!video.participants_in_discussion.includes(participant)) return;
+
+  const participants = {
+    participantsAskingToJoin: video.participants_asking_to_join,
+    participantsInDiscussion: video.participants_in_discussion.filter(
+      (p) => p.id !== participant.id,
+    ),
+  };
+  return await updateLiveParticipants(video, participants);
+};
+
+export const moveParticipantToDiscussion = async (
+  video: Video,
+  participant: Participant,
+) => {
+  if (!video.participants_asking_to_join.includes(participant)) return;
+
+  const participants = {
+    participantsAskingToJoin: video.participants_asking_to_join.filter(
+      (p) => p.id !== participant.id,
+    ),
+    participantsInDiscussion: [
+      ...new Set(video.participants_in_discussion.concat([participant])),
+    ],
```

Comment:
```suggestion
    participantsInDiscussion: video.participants_in_discussion.concat([participant]),
```

Reply:
I added `new Set` to avoid participants duplicates in the list.